Save full response to a file

aldreth opened the pull request:
* Add a new button on the response preview pane
* Save full response to file when button clicked

closes https://github.com/getinsomnia/insomnia/issues/192

gschier commented:
@aldreth I'll take a closer look at this tomorrow but it looks great after a quick glance. 

One thing, since this changes UI, can you attach a screenshot of the new dropdown?

aldreth commented:
NP 😄 
<img width="1280" alt="screen shot 2017-05-14 at 22 15 56" src="https://cloud.githubusercontent.com/assets/389284/26037791/05fbdaf0-38f3-11e7-87bb-02aa8c91ef37.png">

aldreth commented:
@gschier I couldn't see any tests you'd written for saving the body or anything, so didn't try to add any for this.

gschier commented on app/ui/components/dropdowns/preview-mode-dropdown.js:
```diff
@@ -35,6 +35,10 @@ class PreviewModeDropdown extends PureComponent {
           <i className="fa fa-save"/>
           Save to File
```
Should we rename this one to `"Save Request Body"`?

aldreth replied:
Makes sense.

gschier commented on app/ui/components/response-pane.js:
```diff
@@ -81,6 +81,47 @@ class ResponsePane extends PureComponent {
+
+    remote.dialog.showSaveDialog(options, filename => {
+      if (!filename) {
+        trackEvent('Full Response', 'Save Cancel');
```
These `trackEvent` calls should keep the first parameter (category) as `Response`. The "Full" descriptor is better placed in the second parameter (action). 

Eg. `trackEvent('Response', 'Save Full Cancel');`

aldreth replied:
NP

aldreth commented:
@gschier Updated code to reflect your comments. I also spotted that I'd labelled the dropdown as `Save Full Request`, so I've fixed that, and I've removed the dependency of the file extension on the content type, as it doesn't really apply when saving full request.

gschier commented:
Thanks @stewarnet for the extra set of 👀

aldreth commented:
Cheers for the help with this @gschier & for the 👀 @stewarnet.
